### ISSUE TITLE
Reduce package size

### ISSUE THREAD

**Issue by niutech** [intlify/vue-i18n-next#556]
**Is your feature request related to a problem? Please describe.**
The vue-i18n 8.24.5 package weighted 25.9KB minified, whereas version 9.1.6 is 51.3KB minified, which is double the size, according to [Bundlephobia](https://bundlephobia.com/package/vue-i18n@9.1.6). Yet the vue-i18n v9 functionality is not much bigger than in v8.

**Describe the solution you'd like**
Could you please reduce the size of the package, so that is similar to v8?

**Describe alternatives you've considered**
Have a look at e.g. [vue-18n-lite](https://www.npmjs.com/package/vue-i18n-lite), which weights only 1.8KB minified and has no dependencies.

**Additional context**
It is very important for web performance to keep JS bundles as small as possible.

**Comment by kazupon** [comment]
Thank you for your feedback!

I seem that  Bundlephobia is measuring the common.js dist file. 🤔 
According to it,  Bundlephobia is also measuring the size of the compiler it depends on.
vue-i18n-next also provides an unbundled compiler like Vue.
See the docs here for more details.
https://vue-i18n.intlify.dev/installation.html#with-a-bundler

If you build the current vue-i18n-next with `yarn build`, you will get the following output from console.

```
📦  vue-i18n.cjs.prod.js min:54.33kb / gzip:10.49kb / brotli:9.03kb
📦  vue-i18n.esm-browser.prod.js min:39.57kb / gzip:12.76kb / brotli:11.56kb
📦  vue-i18n.global.prod.js min:39.62kb / gzip:12.68kb / brotli:11.54kb
📦  vue-i18n.runtime.esm-browser.prod.js min:26.99kb / gzip:9.02kb / brotli:8.16kb
📦  vue-i18n.runtime.global.prod.js min:27.08kb / gzip:8.96kb / brotli:8.12kb
```

The target output file for the es modules in the browser (without the compiler bundle) is about the same size, just slightly larger than vue-i18n@v8.

And vue-i18n-next provides an option for bundlers to further optimize the size. You can use it to further reduce the size by tree shaking. 
https://vue-i18n.intlify.dev/guide/advanced/optimization.html#reduce-bundle-size-with-feature-build-flags

As far as I can measure, it reduces the overall application size by about 20%.
https://speakerdeck.com/kazupon/state-of-vue-i18n?slide=45

This makes it about the same or more optimized than vue-i18n@v8.x, so I consider it to meet the same requirements for size when compared to vue-i18n@v8.x.

I saw vue-i18n-lite.
It's very simple, small in size, great!
It looks like the interpolation of message is achieved by string substitution.
It seems that message interpolation is achieved by string substitution, which is simple enough for some use cases.

vue-i18n-next can parse message syntax more strictly, based on feedback from users at vue-i18n@v8.x. And AST can be used to optimize performance.
Although vue-i18n-next may look inferior in comparison with vue-i18n-next in terms of build size, it is designed to allow extending the message syntax while maintaining the traditional size.
This is a tradeoff.

However, I understand that web performance is important.
If it is possible to further reduce the size of vue-i18n-next, I will do.

**Comment by niutech** [comment]
Thank you for your explanation.

**Comment by gagarine** [comment]
>  It is very important for web performance to keep JS bundles as small as possible.
No it's not, at least not directly. JS dev have to stop obsessing about size in kb. The interpretation and execution time - not the size - is much more important if you access to a connection >1MB/s .

So does this new version is faster or slower? That's the question. Checking the profilers is a good start.

**Comment by kazupon** [comment]
I've just released v9.2-beta
In that release, vue-i18n lite version is included.
https://github.com/intlify/vue-i18n-next/tree/master/packages/petite-vue-i18n
please check it
thanks!

**Comment by niutech** [comment]
That's great, thank you! The name suggests that it will work with [Petite Vue](https://github.com/vuejs/petite-vue), won't it?
Meanwhile I've made a similar [Petite Vue I18n Lite](https://github.com/niutech/petite-vue-i18n-lite) module for Petite Vue.
